extensions/amp-cxense/0.1/amp-cxense.js

```diff
@@ -0,0 +1,276 @@
+                    self._mpf = self._mpf || mpf;
+                    mpf.video.pause();
+                    resolve();
+                });
+            } else {
+                resolve();
+            }
+        });
+    }
+
+    /** @private */
+    _pauseMpfSync() {
+        return !!(this._mpf && (this._mpf.video.pause() || this._mpf.video.paused));
+    }
+
+    /** @private */
+    _buildWidgetPlaceholder() {
+        const doc = this._getDoc();
+
+        let placeholder = doc.createElement('div');
+        placeholder.className = 'amp-cxense-placeholder';
+
+        let spinner = doc.createElement('div');
+        spinner.className = 'amp-mpf-loader';
+        spinner.appendChild(doc.createElement('div'));
+        spinner.appendChild(doc.createElement('div'));
+        spinner.appendChild(doc.createElement('div'));
+
+        placeholder.appendChild(spinner);
+
+        this.element.appendChild(placeholder);
+        this.applyFillContent(placeholder);
+    }
+
+    /** @private */
+    _injectEmbedScript () {
+        return this._injectScript(this._src);
+    }
+
+    /** @private */
+    _injectScript (src) {
+        const doc = this._getDoc();
+        const script = doc.createElement('script');
+        script.setAttribute("src", src);
+        doc.head.appendChild(script);
+        return loadPromise(script);
+    }
+
+    /** @private */
+    _createChildTarget () {
+        const target = this._getDoc().createElement('div');
+        setStyles(target, {
+            'background': 'black'
+        });
+        let id;
+        let attr;
+        let attributes = Array.prototype.slice.call(this.element.attributes);
+        let shortRegExp = new RegExp('^data-', 'i');
+        let fullRegExp = new RegExp('^data-' + DATA_ATTR_NAME_MIDFIX.replace('^-', ''), 'i');
+
+        while(attr = attributes.pop()) {
+            let name = attr.nodeName;
+            let value = attr.nodeValue;
+
+            let skip = false;
+            if (skipTransfer[name]) {
+                skip = true;
+            }
+            if (name == "id") {
+                value += "-widget";
+            }
+            if (shortRegExp.test(name)) {
+                this.element.removeAttribute(attr.nodeName);
+
+                if (!fullRegExp.test(name)) {
+                    name = _getLongDataAttributeName(name);
+                }
+            }
+            !skip && target.setAttribute(name, value);
+        }
+        this.element.appendChild(target);
+        this._target = target;
+    }
+
+    /** @private */
+    _setCXV () {
+        this._CXV = this.getWin().RAMP;
+    }
+
+    /** @private */
+    _getDoc () {
+        return this.getWin().document;
+    }
+
+    /** @private */
+    _getDataAttribute (name) {
+        name = 'data-' + name.replace(/^data-/i, '');
+        return this.element.getAttribute(name) || this.element.getAttribute(_getLongDataAttributeName(name));
```

Comment:
Please just stick with one attribute naming format. This will cause confusion of doing the same thing. Stick with `data-` attribtues for consistency across AMP.